File: Examples/SimpleEffect/Form1.cs
using System.Windows.Forms;
using SharpDX;
using TDF.Core;
using TDF.Graphics.Cameras;
using TDF.Graphics.Data;
using TDF.Graphics.Effects;
using TDF.Graphics.Models;
using TDF.Graphics.Render;
using TDF.Inputs;

namespace SimpleEffect
{
    public partial class Form1 : Form
    {
        private readonly FreeCamera _freeCamera;
        private  DxModel _model;

        public Form1()
        {
            InitializeComponent();
            _freeCamera = new FreeCamera(new Vector3(0, 25, -150), 0, 0, true);
            DirectX11.Initialize(directXPanel1.Handle, _freeCamera, true);
            var te = new TextureEffect();
            te.InitializeFromFile("Grayscale.fx");
    //        EffectManager.Effects[1] = te;
            _model = GeometryGenerator.GenereateModel<TextureVertex>(GeometryGenerator.CreateGrid(1000, 1000, 20, 20), new Texture("dragonTexture.dds"));

            Shown += (sender, args) => directXPanel1.Run(Update, Draw);
        }

        public void Draw()
        {
            _model.Render();
            label1.Text = directXPanel1.FPS.ToString();
        }

        public new void Update()
        {
            Input.SetMouseCoord(WinAPI.GetCurPos());

            _freeCamera.RotateWithMouse(Input.MouseState);

            if (Input.IsKeyDown(Key.W))
            {
                _freeCamera.DirectionMove(Vector3.ForwardLH);
            }
            else if (Input.IsKeyDown(Key.S))
            {
                _freeCamera.DirectionMove(Vector3.BackwardLH);
            }

            if (Input.IsKeyDown(Key.A))
            {
                _freeCamera.DirectionMove(Vector3.Left);
            }
            else if (Input.IsKeyDown(Key.D))
            {
                _freeCamera.DirectionMove(Vector3.Right);
            }

            _freeCamera.Update();
        }
    }
}

File: VBExample/Form1.vb
Imports TDF.Core
Imports TDF.Inputs
Imports TDF.Graphics.Data
Imports TDF.Graphics.Render
Imports SharpDX
Imports TDF.Graphics.Cameras
Imports TDF.Graphics.Models

Public Class Form1
    ReadOnly _model As DxModel
    ReadOnly _freeCamera As FreeCamera

    Public Sub New()
        InitializeComponent()

        _freeCamera = New FreeCamera(New Vector3(0, 25, -150), 0, 0, True)
        DirectX11.Initialize(DirectXPanel1.Handle, _freeCamera, True)

        _model = GeometryGenerator.GenereateModel(Of TextureVertex)(GeometryGenerator.CreateGrid(1000, 1000, 20, 20), New Texture("dragonTexture.dds"))

    End Sub

    Private Sub Form1_Shown(sender As Object, e As EventArgs) Handles MyBase.Shown
        DirectXPanel1.Run(AddressOf UpdateT, AddressOf DrawT)
    End Sub

    Public Sub UpdateT()
        Input.SetMouseCoord(WinAPI.GetCurPos())

        _freeCamera.RotateWithMouse(Input.MouseState)

        If Input.IsKeyDown(Key.W) Then
            _freeCamera.DirectionMove(Vector3.ForwardLH)
        ElseIf Input.IsKeyDown(Key.S) Then
            _freeCamera.DirectionMove(Vector3.BackwardLH)
        ElseIf Input.IsKeyDown(Key.A) Then
            _freeCamera.DirectionMove(Vector3.Left)
        ElseIf Input.IsKeyDown(Key.D) Then
            _freeCamera.DirectionMove(Vector3.Right)
        End If

        _freeCamera.Update()
    End Sub

    Public Sub DrawT()
        _model.Render()
    End Sub
End Class
